install_requirements.py

```diff
@@ -0,0 +1,153 @@
+import os
+import platform
+import re
+import sys
+import subprocess
+
+# Before doing anything, cd to the directory containing this script.
+os.chdir(os.path.dirname(os.path.abspath(__file__)))
+
+# Find the names of the python tools to use.
+PYTHON_EXECUTABLE = os.getenv('PYTHON_EXECUTABLE')
+
+if not PYTHON_EXECUTABLE:
+    CONDA_DEFAULT_ENV = os.getenv('CONDA_DEFAULT_ENV')
+    if not CONDA_DEFAULT_ENV or CONDA_DEFAULT_ENV == "base" or not subprocess.call(["which", "python"]):
+        PYTHON_EXECUTABLE = "python3"
+    else:
+        PYTHON_EXECUTABLE = "python"
+
+if PYTHON_EXECUTABLE == "python":
+    PIP_EXECUTABLE = "pip"
+else:
+    PIP_EXECUTABLE = "pip3"
```

Comment:
This is only necessary in the bash script, since we needed to know which python binary to use when invoking this script. Once we're in the script, `sys.executable` will be the path to the python binary used to invoke it.

Once in the script, it'd be better to run pip as `{sys.executable} -m pip <pip-args>`. Then we know it's using the same python binary, and we don't need to treat `pip` as a special case, so no need for PIP_EXECUTABLE.